src/overlays/actors/ovl_Obj_Mine/z_obj_mine.c

```diff
@@ -24,131 +53,1180 @@ ActorInit Obj_Mine_InitVars = {
+        posDiff.y = sphere->center.y - attackActor->world.pos.y;
+        posDiff.z = sphere->center.z - attackActor->world.pos.z;
+
+        if (!ObjMine_GetUnitVec3f(&posDiff, knockbackDir)) {
+            Math_Vec3f_Copy(knockbackDir, &sStandardBasis.y);
+        }
+    }
+}
+
+void ObjMine_AirWater_Noop(ObjMine* this) {
+}
+
+void ObjMine_ReplaceTranslation(Vec3f* translation) {
+    MtxF* matrix = Matrix_GetCurrent();
+
+    matrix->mf[3][0] = translation->x;
+    matrix->mf[3][1] = translation->y;
+    matrix->mf[3][2] = translation->z;
+}
+
+void ObjMine_ReplaceRotation(ObjMineMtxF3* basis) {
+    MtxF* matrix = Matrix_GetCurrent();
+
+    matrix->mf[0][0] = basis->x.x;
+    matrix->mf[0][1] = basis->x.y;
+    matrix->mf[0][2] = basis->x.z;
+    matrix->mf[0][3] = 0.0f;
+
+    matrix->mf[1][0] = basis->y.x;
+    matrix->mf[1][1] = basis->y.y;
+    matrix->mf[1][2] = basis->y.z;
+    matrix->mf[1][3] = 0.0f;
+
+    matrix->mf[2][0] = basis->z.x;
+    matrix->mf[2][1] = basis->z.y;
+    matrix->mf[2][2] = basis->z.z;
+    matrix->mf[2][3] = 0.0f;
+
+    matrix->mf[3][0] = 0.0f;
+    matrix->mf[3][1] = 0.0f;
+    matrix->mf[3][2] = 0.0f;
+    matrix->mf[3][3] = 1.0f;
+}
+
+s32 ObjMine_StepUntilParallel(Vec3f* value, Vec3f* target, f32 angleStep) {
+    Vec3f perpVec;
+    Vec3f prevValue;
+    Vec3f perpNormal;
+    f32 cosAngle = Math3D_Parallel(value, target);
+
+    if (cos_rad(angleStep) <= cosAngle) {
+        Math_Vec3f_Copy(value, target);
+        return true;
+    }
+
+    Matrix_Push();
+    Math_Vec3f_Copy(&prevValue, value);
+    Math3D_CrossProduct(value, target, &perpVec);
+    if (ObjMine_GetUnitVec3f(&perpVec, &perpNormal)) {
+        Matrix_RotateAxisS(RADF_TO_BINANG(angleStep), &perpNormal, MTXMODE_NEW);
+        Matrix_MultVec3f(&prevValue, value);
+    } else {
+        Matrix_RotateXFNew(angleStep);
+        Matrix_MultVec3f(&prevValue, value);
+    }
+
+    Matrix_Pop();
+    return false;
+}
+
+void ObjMine_UpdateCollider(ObjMine* this) {
+    this->collider.elements[0].dim.worldSphere.center.x = this->actor.world.pos.x;
+    this->collider.elements[0].dim.worldSphere.center.y = this->actor.world.pos.y;
+    this->collider.elements[0].dim.worldSphere.center.z = this->actor.world.pos.z;
+}
+
+void ObjMine_Air_InitChain(ObjMine* this, s32 linkCount) {
+    f32 linkCountF = (f32)linkCount;
+    ObjMineAirChain* airChain = &this->chain.air;
+    ObjMineAirLink* airLink;
+    s32 i;
+    f32 wallCheckRadius = this->actor.home.rot.z * 5.0f;
+
+    airChain->basis.x.x = 1.0f;
+    airChain->basis.y.y = 1.0f;
+    airChain->basis.z.z = 1.0f;
+
+    airChain->translation.y = 1.0f;
+
+    // Sets restoring force and drag. Longer chains have lower frequency and less drag.
+    if (linkCount > 0) {
+        airChain->restore = -sqrtf((0.124992f / OBJMINE_CHAIN_MAX) / linkCountF);  // constant is close to 1/8
+        airChain->drag = 0.95f + ((0.02000004f / OBJMINE_CHAIN_MAX) * linkCountF); // constant is close to 1/50
```

Comment:
Hmm do 
`(1.0f / 0.001f) / 8000.0f` or maybe `1000.0f / 8000.0f `
`(20.0f * 0.001f)`
work for the constants?

Reply:
These are both way too far off to be a float calculation error. More likely the devs truncated the decimal expansions early.